### PR TITLE
fix: Fix security issues with event blobs

### PR DESCRIPTION
**Pull request by sadhansood**
## Description

Fixes issues related to being able to increment certified checkpoint incorrectly and causing OOM by repeatedly attesting invalid blob ids.
## Test plan

Added test cases written by Ben

### CODE REVIEW

File: contracts/walrus/tests/system/event_blob_tests.move [hunk 1]
```diff
@@ -224,6 +224,119 @@ public fun test_event_blob_certify_change_epoch() {
     destroy(system);
 }
 
+#[test]
+public fun test_certify_invalid_blob_id() {
```
Comment (benr-ml) [inline contracts/walrus/tests/system/event_blob_tests.move:220]: let's fix format and clean the tests a bit

File: contracts/walrus/sources/system/system_state_inner.move [hunk 2]
```diff
@@ -430,7 +430,9 @@ public(package) fun certify_event_blob(
             ending_checkpoint_sequence_num,
             blob_id,
         );
-    self.event_blob_certification_state.stop_tracking_blob(blob_id);
```
Comment (karlwuest) [inline contracts/walrus/sources/system/system_state_inner.move:433]: Suggestion: I think the `stop_tracking_blob` function is now no longer needed. Can you remove that function?

Comment (sadhansood) [reply contracts/walrus/sources/system/system_state_inner.move:433]: done

File: contracts/walrus/sources/system/event_blob.move [hunk 3]
```diff
@@ -24,8 +24,7 @@ public struct EventBlob has copy, drop, store {
 
 /// State of event blob stream.
 #[allow(unused_field)]
```
Comment (karlwuest) [inline contracts/walrus/sources/system/event_blob.move]: I think the `#[allow(unused_field)]` can be removed now. You'll probably also need to change the structs in rust (with the `mainnet-contracts` feature), to make sure the deserialization works (this is probably why the rust tests are failing).

**Pull request merged**